### PR TITLE
Rename nonnative to emulated, as in `r1cs-std`

### PR DESCRIPTION
**Pull request by autquis**
Just a refactor according to the last changes in `r1cs-std`

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,8 +31,8 @@ debug = true
 ark-ff = { git = "https://github.com/arkworks-rs/algebra/" }
 ark-ec = { git = "https://github.com/arkworks-rs/algebra/" }
 ark-serialize = { git = "https://github.com/arkworks-rs/algebra/" }
-ark-crypto-primitives = { git = "https://github.com/arkworks-rs/crypto-primitives" }
```
Comment (mmagician) [inline Cargo.toml:34]: this will still have to change back once the rest of the deps are updated

Comment (autquis) [reply Cargo.toml:34]: Of course! 👍🏻

**Comment by autquis** [comment]
The CI fails. It is probably due to the instability of the nightly build, though.
I checked the warning. There is no dead code, and the stable build checks pass.

**Comment by mmagician** [comment]
This is a private field, so it it's never read, I think it's alright to remove already here.

**Comment by autquis** [comment]
Thanks. I removed it. Also, I think further refactoring is possible but I left it for another PR later.

File: poly-commit/src/streaming_kzg/data_structures.rs [hunk 2]
```diff
@@ -140,7 +140,7 @@ where
 
 /// Stream implementation of foleded polynomial.
 #[derive(Clone, Copy)]
-pub struct FoldedPolynomialStream<'a, F, S>(FoldedPolynomialTree<'a, F, S>, usize);
+pub struct FoldedPolynomialStream<'a, F, S>(FoldedPolynomialTree<'a, F, S>);
```
Comment (Pratyush) [inline poly-commit/src/streaming_kzg/data_structures.rs:143]: cc @mmaker is it okay to delete the field?

Comment (mmaker) [reply poly-commit/src/streaming_kzg/data_structures.rs:143]: yes I think so! I think `len` was used at the beginning when I thought we needed to provide shorter length than 2^|challenges| but in the end was not used. I'll fix it if new errors come up.

**Review by Pratyush — APPROVED** [review]

File: poly-commit/src/streaming_kzg/data_structures.rs (hunk 2 above)
Comment (mmagician) [reply poly-commit/src/streaming_kzg/data_structures.rs:143]: Thanks!

**Pull request merged**